habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py

```diff
@@ -169,14 +184,33 @@ def _sample_active(self):
         assert not self._pop_config.self_play_batched
 
         # Random sample over which agents are active.
-        self._active_agents = np.random.choice(
-            self._all_agent_idxs,
-            size=self._pop_config.num_active_agents,
-            replace=self._pop_config.allow_self_play,
-        )
+        prev_num_agents = 0
+        active_agents = []
+        active_agent_types = []
+        for agent_type_ind in range(self._pop_config.num_agent_types):
```

Comment:
How often is this method called? I don't expect sampling of active agents to happen a lot but if that is the case, you should consider caching the _pop_config for efficiency.

Reply:
Sorry, what is it that we should precompute or cache? The _pop_config is already cached right? And the rest of the code is doing random sampling which needs to be called every time.